source/common/http/filter/ratelimit.cc

```diff
@@ -16,6 +17,13 @@ const std::unique_ptr<const Http::HeaderMap> Filter::TOO_MANY_REQUESTS_HEADER{
         {Http::Headers::get().Status, std::to_string(enumToInt(Code::TooManyRequests))}}};
 
 void Filter::initiateCall(const HeaderMap& headers) {
+  bool is_internal_request = Http::Utility::isInternalRequest(headers);
```

Comment:
Check for presence of EnvoyInternalRequest header directly, to avoid parsing again.